system/density/buffers/buffer.c

```diff
@@ -0,0 +1,184 @@
+/*
```

Comment:
Is there a reason you're adding the Density sources directly to our source tree, instead of making it a git submodule ?